avm/res/network/p2s-vpn-gateway/main.bicep

```diff
@@ -0,0 +1,192 @@
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource p2sVpnGateway 'Microsoft.Network/p2svpnGateways@2024-01-01' = {
+  name: name
+  location: location
+  tags: tags
+  properties: {
+    customDnsServers: customDnsServers
+    isRoutingPreferenceInternet: isRoutingPreferenceInternet
+    p2SConnectionConfigurations: [
+      {
+        name: p2SConnectionConfigurationsName
+        properties: {
+          enableInternetSecurity: enableInternetSecurity
+          routingConfiguration: {
+            associatedRouteTable:  {
+              id: resourceId('Microsoft.Network/virtualHubs/hubRouteTables','${virtualHubName}','${associatedRouteTableName}')
+            }
+            inboundRouteMap: (!empty(inboundRouteMapResourceId)) ? {
+              id:  inboundRouteMapResourceId
+            } : null
+            outboundRouteMap: (!empty(outboundRouteMapResourceId)) ? {
+              id: outboundRouteMapResourceId
+            } : null
+            propagatedRouteTables: {
+              ids: [
+                for table in (propagatedRouteTableNames): {
+                  id: resourceId('Microsoft.Network/virtualHubs/hubRouteTables','${virtualHubName}','${table}')
+                }
+              ]
+              labels: propagatedLabelNames
+            }
+          }
+          vpnClientAddressPool: {
+            addressPrefixes: vpnClientAddressPoolAddressPrefixes
+          }
+        }
+      }
+    ]
+    virtualHub: {
+      id: virtualHubId
+    }
+    vpnGatewayScaleUnit: vpnGatewayScaleUnit
+    vpnServerConfiguration: {
+      id: vpnServerConfigurationId
+    }
+  }
+}
+
+resource vpnGateway_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete'
+      ? 'Cannot delete resource or child resources.'
+      : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: p2sVpnGateway
+}
+
+@description('The name of the user VPN configuration.')
+output name string = p2sVpnGateway.name
+
+@description('The resource ID of the user VPN configuration.')
+output resourceId string = p2sVpnGateway.id
+
+@description('The name of the resource group the user VPN configuration was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The location the resource was deployed into.')
+output location string = p2sVpnGateway.location
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. Specify the type of lock.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
+}?
+
+type vnetRoutesStaticRoutesType = {
```

Comment:
This is not used. Should be type be removed?

Reply:
It should be (used); somehow I'm missing a parameter... fixing now.